plugins/mongodb/mongodbembed.cpp

```diff
@@ -0,0 +1,2302 @@
+    }
+
+    /**
+     * @brief Binds an ECL Boolean param to a bsoncxx::types::b_bool.
+     * 
+     * @param name Name of the parameter.
+     * @param val Boolean value.
+     */
+    void MongoDBEmbedFunctionContext::bindBooleanParam(const char *name, bool val)
+    {
+        checkNextParam(name);
+        std::string terminatedString = name;
+
+        query->build()->append(kvp(terminatedString, bsoncxx::types::b_bool{val}));
+    }
+
+    /**
+     * @brief Binds an ECL Size param to a bsoncxx::types::b_int64.
+     * 
+     * @param name Name of the parameter.
+     * @param size Size of the value.
+     * @param val Integer value.
+     */
+    void MongoDBEmbedFunctionContext::bindSignedSizeParam(const char *name, int size, __int64 val)
+    {
+        bindSignedParam(name, val);
+    }
+    
+    /**
+     * @brief Binds an ECL Unsigned Size param to a bsoncxx::types::b_int64.
+     * 
+     * @param name Name of the parameter.
+     * @param size Size of the value.
+     * @param val Integer value.
+     */
+    void MongoDBEmbedFunctionContext::bindUnsignedSizeParam(const char *name, int size, unsigned __int64 val)
+    {
+        bindUnsignedParam(name, val);
+    }
+     
+    /**
+     * @brief Binds an ECL Real4 param to a bsoncxx::types::b_double.
+     * 
+     * @param name Name of the parameter.
+     * @param val float value.
+     */
+    void MongoDBEmbedFunctionContext::bindFloatParam(const char *name, float val)
+    {
+        checkNextParam(name);
+        query->build()->append(kvp(std::string(name), bsoncxx::types::b_double{val}));
+    }
+     
+    /**
+     * @brief Binds an ECL Real param to a bsoncxx::types::b_double.
+     * 
+     * @param name Name of the parameter.
+     * @param val Double value.
+     */
+    void MongoDBEmbedFunctionContext::bindRealParam(const char *name, double val)
+    {
+        checkNextParam(name);
+        query->build()->append(kvp(std::string(name), bsoncxx::types::b_double{val}));
+    }
+    
+    /**
+     * @brief Binds an ECL Integer param to a bsoncxx::types::b_int64.
+     * 
+     * @param name Name of the parameter.
+     * @param val Signed Integer value.
+     */
+    void MongoDBEmbedFunctionContext::bindSignedParam(const char *name, __int64 val)
+    {
+        checkNextParam(name);
+        int64_t mongoVal = val;
+        std::string terminatedString = name;
+
+        query->build()->append(kvp(terminatedString, bsoncxx::types::b_int64{mongoVal}));
+    }
+
+    /**
+     * @brief Binds an ECL Unsigned Integer param to a bsoncxx::types::b_int64.
+     * 
+     * @param name Name of the parameter.
+     * @param val Unsigned Integer value.
+     */
+    void MongoDBEmbedFunctionContext::bindUnsignedParam(const char *name, unsigned __int64 val)
+    {
+        checkNextParam(name);
+        int64_t mongoVal = val;
+        std::string terminatedString = name;
+
+        query->build()->append(kvp(terminatedString, bsoncxx::types::b_int64{mongoVal}));
```

Comment:
trivial, but likely a little cleaner to create the std::string inline as was done in other functions:
query->build()->append(kvp(std::string(name)

Reply:
Changed to inline.